### ISSUE TITLE
"not found: value usePipelining"

### ISSUE THREAD

**Issue by eed3si9n** [sbt/sbt#5750]
## steps

This happens on sbt/sbt using freshly built snapshot:

```scala
ThisBuild / usePipelining := true
```

## problem

```scala
ThisBuild / usePipelining := true
            ^
[error] Type error in expression
[warn] Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? (default: r)
```

## expectation

It should load.

**Comment by eed3si9n** [comment]
Similar to https://github.com/sbt/sbt/issues/5915, this can be worked around with `ThisBuild / useCoursier := false` in `project/plugins.sbt`.

**Comment by eatkins** [comment]
I just ran into this again. I wonder if a workaround on the sbt side could be to prepend the plugin classpath with the metabuild sbt jars or change the parent classloader for Eval to be the metabuild classloader.

**Comment by eatkins** [comment]
I verified that this diff is a workaround:
```patch
diff --git a/main/src/main/scala/sbt/internal/Load.scala b/main/src/main/scala/sbt/internal/Load.scala
index 2e40d7541..0ec298e6f 100755
--- a/main/src/main/scala/sbt/internal/Load.scala
+++ b/main/src/main/scala/sbt/internal/Load.scala
@@ -432,8 +432,12 @@ private[sbt] object Load {
     mkEval(defs.target ++ plugs.classpath, defs.base, options)
 
   def mkEval(classpath: Seq[File], base: File, options: Seq[String]): Eval = {
-    if (classpath.toString.contains("sbt-1.3.0.jar")) new Exception("").printStackTrace(System.err)
-    new Eval(options, classpath, s => new ConsoleReporter(s), Some(evalOutputDirectory(base)))
+    val metaBuildClasspath = getClass.getClassLoader match {
+      case u: java.net.URLClassLoader => u.getURLs.map(u => new File(u.toURI.getPath))
+      case _                          => Array.empty[File]
+    }
+    val fullClasspath = metaBuildClasspath ++ classpath
+    new Eval(options, fullClasspath, s => new ConsoleReporter(s), Some(evalOutputDirectory(base)))
   }

```
It would be much better to fix the issue in https://github.com/coursier/coursier/issues/1877 though

**Comment by eed3si9n** [comment]
> I wonder if a workaround on the sbt side could be to prepend the plugin classpath with the metabuild sbt jars or change the parent classloader for Eval to be the metabuild classloader.

My idea in https://github.com/sbt/sbt/issues/4437 was to reuse the sbt JARs resolved already by the launcher. I got it to work with Ivy in https://github.com/sbt/sbt/pull/4443, but I had to roll it back because Coursier's exclusion was buggy. It might be worth revisiting since it could potentially fix the double resolution issue the first time someone uses sbt.
